xds/internal/resolver/serviceconfig.go

```diff
@@ -121,7 +133,11 @@ type routeCluster struct {
 
 type route struct {
 	m                 *xdsresource.CompositeMatcher // converted from route matchers
+
+	// Exactly one of clusterSpecifierPlugin or clusters will be set.
+    clusterSpecifierPlugin string
```

Comment:
Can we put the CSP name into clusters instead so we don't need another field?

Reply:
I see what your saying (persist as little state as possible), but I much prefer keeping this as a separate field. Adding csp to clusters would conflate an individual csp name to a WRR type which can hold onto many cluster names and chooses one randomly. This keeps the branching logic (two fields in state, two separate ways of persistence/handling), which started from the xdsclient.

Reply:
It's not about "persisting state", it's about reducing & simplifying code and data structures.  We use the same code paths already for when WRR is used vs. when a single cluster is specified directly, so why not fold CSP into it as well?  In gRPC, they are equivalent: CSPs are the same as ordinary clusters from the name resolver and cluster manager LB policy's perspectives.

Reply:
Kept the branching in newConfigSelector (for prefix + hardcoded weight 1 logic, which was in unmarshal_rds for singular cluster)...but now it's one codepath in SelectConfig/one piece of state clusters.